exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/traces/SpanStatelessMarshaler.java

```diff
@@ -0,0 +1,122 @@
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+final class SpanStatelessMarshaler implements StatelessMarshaler<SpanData> {
+  static final SpanStatelessMarshaler INSTANCE = new SpanStatelessMarshaler();
+  private static final byte[] EMPTY_BYTES = new byte[0];
+
+  @Override
+  public void writeTo(Serializer output, SpanData span, MarshalerContext context)
+      throws IOException {
+    output.serializeTraceId(Span.TRACE_ID, span.getTraceId(), context);
+    output.serializeSpanId(Span.SPAN_ID, span.getSpanId(), context);
+
+    byte[] traceStateUtf8 = context.getData(byte[].class);
+    output.serializeString(Span.TRACE_STATE, traceStateUtf8);
+    String parentSpanId =
+        span.getParentSpanContext().isValid() ? span.getParentSpanContext().getSpanId() : null;
+    output.serializeSpanId(Span.PARENT_SPAN_ID, parentSpanId, context);
+
+    output.serializeString(Span.NAME, span.getName(), context);
+    output.serializeEnum(Span.KIND, toProtoSpanKind(span.getKind()));
+
+    output.serializeFixed64(Span.START_TIME_UNIX_NANO, span.getStartEpochNanos());
+    output.serializeFixed64(Span.END_TIME_UNIX_NANO, span.getEndEpochNanos());
+
+    output.serializeRepeatedMessage(
+        Span.ATTRIBUTES, span.getAttributes(), KeyValueStatelessMarshaler.INSTANCE, context);
+    int droppedAttributesCount = span.getTotalAttributeCount() - span.getAttributes().size();
+    output.serializeUInt32(Span.DROPPED_ATTRIBUTES_COUNT, droppedAttributesCount);
+
+    output.serializeRepeatedMessage(
+        Span.EVENTS, span.getEvents(), SpanEventStatelessMarshaler.INSTANCE, context);
+    int droppedEventsCount = span.getTotalRecordedEvents() - span.getEvents().size();
+    output.serializeUInt32(Span.DROPPED_EVENTS_COUNT, droppedEventsCount);
+
+    output.serializeRepeatedMessage(
+        Span.LINKS, span.getLinks(), SpanLinkStatelessMarshaler.INSTANCE, context);
+    int droppedLinksCount = span.getTotalRecordedLinks() - span.getLinks().size();
+    output.serializeUInt32(Span.DROPPED_LINKS_COUNT, droppedLinksCount);
+
+    output.serializeMessage(
+        Span.STATUS, span.getStatus(), SpanStatusStatelessMarshaler.INSTANCE, context);
+
+    output.serializeFixed32(
+        Span.FLAGS,
+        SpanFlags.withParentIsRemoteFlags(
+            span.getSpanContext().getTraceFlags(), span.getParentSpanContext().isRemote()));
+  }
+
+  @Override
+  public int getBinarySerializedSize(SpanData span, MarshalerContext context) {
+    int size = 0;
+    size += MarshalerUtil.sizeTraceId(Span.TRACE_ID, span.getTraceId());
+    size += MarshalerUtil.sizeSpanId(Span.SPAN_ID, span.getSpanId());
+
+    TraceState traceState = span.getSpanContext().getTraceState();
+    byte[] traceStateUtf8 =
+        traceState.isEmpty()
+            ? EMPTY_BYTES
+            : encodeTraceState(traceState).getBytes(StandardCharsets.UTF_8);
+    context.addData(traceStateUtf8);
```

Comment:
Consider a utility method for this since its also used in `SpanLinkStatelessMarshaler`.